fix link to async return types spec, clarify Task-like return types. (#23386)
dotnet/docs: docs/csharp/programming-guide/concepts/async/async-return-types.md

.NET provides the <xref:System.Threading.Tasks.ValueTask%601?displayProperty=nameWithType> structure as a lightweight implementation of a generalized task-returning value. To use the <xref:System.Threading.Tasks.ValueTask%601?displayProperty=nameWithType> type, you must add the `System.Threading.Tasks.Extensions` NuGet package to your project. The following example uses the <xref:System.Threading.Tasks.ValueTask%601> structure to retrieve the value of two dice rolls.

:::code language="csharp" source="snippets/async-return-types/async-valuetask.cs":::

> Writing a generalized async return type is an advanced scenario. You should use `Task`, `Task<T>` and `ValueTask<T>`. Those types cover most scenarios for asynchronous code.

Reviewer: ```suggestion
Writing a generalized async return type is an advanced scenario. Consider using the `Task`, `Task<T>` and `ValueTask<T>` types instead, which cover most scenarios for asynchronous code.
```
